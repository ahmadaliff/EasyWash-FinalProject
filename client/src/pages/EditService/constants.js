export const EDIT_SERVICE = 'services/EDIT_SERVICE';
export const GET_SERVICE = 'services/GET_SERVICE';
export const SET_SERVICE = 'services/SET_SERVICE';
export const RESET_SERVICE = 'services/RESET_SERVICE';
