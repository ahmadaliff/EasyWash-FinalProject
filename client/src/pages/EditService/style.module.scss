@import '@styles/base';

.mainWrap {
  @include directionY;
  padding: 1rem;
  min-height: var(--min-height);
  @include apply-at-md-size {
    min-height: var(--min-height-sm);
    padding: 2rem 5rem 2rem 5rem;
  }
  align-items: center;
  box-sizing: border-box;
  position: relative;
  * {
    box-sizing: border-box;
  }
  .header {
    @include directionX;
    gap: 1rem;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    .backButton {
      background-color: var(--color-primary);
      border-radius: 10rem;
      padding: 0.25rem 1rem 0.25rem 1rem;
      text-transform: capitalize;
      font-size: medium;
    }
  }
  .cardWrap {
    @include directionY;
    gap: 0.5rem;
    width: 100%;
    box-shadow: none;
    @include apply-at-md-size {
      box-shadow: 0 0 0.25rem gray;
    }
    .cardContent {
      .buttonEditProfile {
        width: 100%;
        background-color: var(--color-primary);
        color: var(--color-bg);
        border: none;
        border-radius: 2.5rem;
        padding: 0.5rem 2.5rem 0.5rem 2.5rem;
        margin: auto;
        margin-top: 0.75rem;
        box-shadow: none;
        font-weight: bold;
        cursor: pointer;
        &:hover {
          transform: scale(1.025, 1.025);
        }
      }
      .inputWrap {
        @include directionX;
        align-items: start;
        gap: 1rem;
        .inputLabel {
          padding: 8px 0 8px 0;
          @include apply-at-md-size {
            padding: 10px 0 10px 0;
          }
          font-size: small;
          .inputLabelTitle {
            margin: 0;
            font-weight: 700;
          }
          float: left;
          width: 10rem;
        }
        .input {
          width: 100%;
        }
      }
    }
  }
}
.profileInfo {
  @include directionX;
  align-items: center;
  font-size: 0.75rem;
  font-weight: 500;
  .profileLabel {
    float: left;
    width: 10rem;
  }
  .profileText {
    width: 100%;
  }
}
