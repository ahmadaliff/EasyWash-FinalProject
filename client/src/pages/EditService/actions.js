import { EDIT_SERVICE, GET_SERVICE, RESET_SERVICE, SET_SERVICE } from './constants';

export const actionEditService = (id, data, callback) => ({
  type: EDIT_SERVICE,
  id,
  data,
  callback,
});

export const actionGetService = (id) => ({
  type: GET_SERVICE,
  id,
});

export const actionSetService = (data) => ({
  type: SET_SERVICE,
  data,
});

export const actionResetService = () => ({
  type: RESET_SERVICE,
});
