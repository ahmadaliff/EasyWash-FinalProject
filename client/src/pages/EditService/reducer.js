import { produce } from 'immer';
import { RESET_SERVICE, SET_SERVICE } from './constants';

export const initialState = {
  service: null,
};

const editServiceReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case SET_SERVICE:
        draft.service = action.data;
        break;
      case RESET_SERVICE:
        return initialState;
    }
  });

export default editServiceReducer;
