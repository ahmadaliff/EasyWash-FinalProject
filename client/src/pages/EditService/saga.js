import toast from 'react-hot-toast';
import { call, put, takeLatest } from 'redux-saga/effects';

import { apiEditServices, apiGetService } from '@domain/api';

import intlHelper from '@utils/intlHelper';

import { EDIT_SERVICE, GET_SERVICE } from '@pages/EditService/constants';

import { actionSetService } from '@pages/EditService/actions';
import { setLoading, showPopup } from '@containers/App/actions';

function* sagaEditService({ id, data, callback }) {
  yield put(setLoading(true));
  try {
    const response = yield call(apiEditServices, data, id);
    toast.success(intlHelper({ message: response.message }));
    yield call(callback);
  } catch (error) {
    if (error?.response?.status === 400 || error?.response?.status === 404) {
      toast.error(intlHelper({ message: error.response.data.message }));
    } else {
      yield put(showPopup());
    }
  }
  yield put(setLoading(false));
}

function* sagaGetService({ id }) {
  yield put(setLoading(true));
  try {
    const response = yield call(apiGetService, id);
    yield put(actionSetService(response.data));
  } catch (error) {
    if (error?.response?.status === 400 || error?.response?.status === 404) {
      toast.error(intlHelper({ message: error.response.data.message }));
    } else {
      yield put(showPopup());
    }
  }
  yield put(setLoading(false));
}

export default function* editServiceSaga() {
  yield takeLatest(EDIT_SERVICE, sagaEditService);
  yield takeLatest(GET_SERVICE, sagaGetService);
}
