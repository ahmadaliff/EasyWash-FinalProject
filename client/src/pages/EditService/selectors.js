import { createSelector } from 'reselect';

import { initialState as editServiceInitialState } from '@pages/EditService/reducer';

const selectServiceState = (state) => state.editService || editServiceInitialState;
export const selectService = createSelector(selectServiceState, (state) => state.service);
