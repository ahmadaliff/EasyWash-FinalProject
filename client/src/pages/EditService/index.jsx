import PropTypes from 'prop-types';
import { useForm } from 'react-hook-form';
import { useEffect, useState } from 'react';
import { useNavigate, useParams } from 'react-router-dom';
import { connect, useDispatch } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { FormattedMessage, injectIntl } from 'react-intl';

import { Button, Card, CardContent, FormControl, Switch } from '@mui/material';
import { ArrowBack } from '@mui/icons-material';

import InputRHF from '@components/InputRHF';

import { actionEditService, actionGetService, actionResetService } from '@pages/EditService/actions';
import { selectService } from '@pages/EditService/selectors';

import classes from '@pages/EditService/style.module.scss';

const EditService = ({ service, intl: { formatMessage } }) => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const { id } = useParams();
  const [isUnit, setIsUnit] = useState(false);
  const {
    handleSubmit,
    register,
    formState: { errors },
  } = useForm();

  useEffect(() => {
    if (!service) dispatch(actionGetService(id));
    else setIsUnit(!!service.isUnit);
    return () => {
      if (service) dispatch(actionResetService());
    };
  }, [dispatch, id, service]);

  const onSubmit = (data) => {
    delete data.name;
    data.isUnit = isUnit;
    dispatch(
      actionEditService(service.id, data, () => {
        setTimeout(() => {
          navigate(-1);
        }, 1000);
      })
    );
  };

  return (
    <main className={classes.mainWrap} data-testid="editservice-page">
      <div className={classes.header}>
        <h3>
          <FormattedMessage id="app_service_edit_header" />
        </h3>
        <Button
          variant="contained"
          className={classes.backButton}
          startIcon={<ArrowBack />}
          onClick={() => navigate(-1)}
          data-testid="button-back"
        >
          <FormattedMessage id="app_back" />
        </Button>
      </div>
      <Card className={classes.cardWrap}>
        <CardContent className={classes.cardContent}>
          <form onSubmit={handleSubmit(onSubmit)}>
            <div className={classes.inputWrap}>
              <div className={classes.inputLabel}>
                <p className={classes.inputLabelTitle}>
                  <FormattedMessage id="app_service_name" />
                </p>
                <p>
                  <FormattedMessage
                    id="app_service_name_desc"
                    values={{
                      // eslint-disable-next-line react/no-unstable-nested-components
                      b: (chunks) => <b>{chunks}</b>,
                    }}
                  />
                </p>
              </div>
              <InputRHF
                input={{
                  name: 'name',
                  type: 'text',
                  label: formatMessage({ id: 'app_service_name' }),
                  value: service?.name,
                }}
                disabled
                register={register}
                errors={errors}
              />
            </div>
            <div className={classes.inputWrap}>
              <div className={classes.inputLabel}>
                <p className={classes.inputLabelTitle}>
                  <FormattedMessage id="app_service_price" />
                </p>
                <p>
                  <FormattedMessage id="app_service_price_desc" />
                </p>
              </div>
              <InputRHF
                input={{
                  name: 'price',
                  required: formatMessage({ id: 'app_service_price_require_message' }),
                  type: 'number',
                  label: formatMessage({ id: 'app_service_price' }),
                  value: service?.price.toString(),
                }}
                register={register}
                errors={errors}
              />
            </div>
            <div className={classes.inputWrap}>
              <div className={classes.inputLabel}>
                <p className={classes.inputLabelTitle}>
                  <FormattedMessage id="app_service_isUnit" />
                </p>
                <p>
                  <FormattedMessage id="app_service_isUnit_desc" />
                </p>
              </div>
              <FormControl fullWidth>
                {service && (
                  <Switch
                    onChange={(e) => setIsUnit(e.target.checked)}
                    checked={!!service?.isUnit}
                    data-testid="isUnit"
                  />
                )}
              </FormControl>
            </div>
            <button type="submit" className={classes.buttonEditProfile} data-testid="button-submit">
              <FormattedMessage id="app_service_edit_header" />
            </button>
          </form>
        </CardContent>
      </Card>
    </main>
  );
};

EditService.propTypes = {
  intl: PropTypes.object,
  service: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  service: selectService,
});

export default injectIntl(connect(mapStateToProps)(EditService));
