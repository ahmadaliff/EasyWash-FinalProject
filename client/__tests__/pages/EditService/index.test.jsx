import { render, fireEvent, screen } from '@utils/testHelper';
import EditService from '@pages/EditService';

let wrapper;
const mockDispatch = jest.fn();
const mockNavigate = jest.fn();
const mockUseParams = jest.fn().mockReturnValue({ id: '1' });

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
  useParams: () => mockUseParams,
}));
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
}));
jest.mock('@pages/EditService/selectors', () => ({
  selectService: () => ({
    id: 1,
    name: 'Service Name',
    price: 50,
    isUnit: true,
  }),
}));

const mockProps = {
  intl: {
    formatMessage: jest.fn(),
  },
};

beforeEach(() => {
  wrapper = render(<EditService {...mockProps} />);
  jest.clearAllMocks();
});

describe('EditService Page', () => {
  test('Correct render', async () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('editservice-page')).toBeInTheDocument();
  });

  test('Should be navigate -1', () => {
    const { getByTestId } = wrapper;
    fireEvent.click(getByTestId('button-back'));
    expect(mockNavigate).toHaveBeenCalledWith(-1);
  });
  test('Should change field value', () => {
    const inputName = screen.getByLabelText('Nama Layanan');
    const inputPrice = screen.getByLabelText('Harga');
    fireEvent.change(inputName, { target: { value: 'Updated Name' } });
    fireEvent.change(inputPrice, { target: { value: 2000 } });
    expect(inputName.value).toBe('Updated Name');
    expect(inputPrice.value).toBe('2000');
  });
  test('Should submit', () => {
    const { getByTestId } = wrapper;
    const buttonSubmit = getByTestId('button-submit');
    fireEvent.click(buttonSubmit);
  });

  test('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
