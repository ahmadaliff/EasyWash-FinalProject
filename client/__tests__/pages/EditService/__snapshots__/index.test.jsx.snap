// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditService Page Matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="mainWrap"
        data-testid="editservice-page"
      >
        <div
          class="header"
        >
          <h3>
            Ubah Layanan
          </h3>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backButton css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-back"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowBackIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
            </span>
            Kembali
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root cardWrap css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root cardContent css-46bh2p-MuiCardContent-root"
          >
            <form>
              <div
                class="inputWrap"
              >
                <div
                  class="inputLabel"
                >
                  <p
                    class="inputLabelTitle"
                  >
                    Nama Layanan
                  </p>
                  <p>
                    Nama layanan 
Disarankan untuk tidak menggunakan huruf kapital berlebih, memasukkan lebih dari 1 layanan, dan kata-kata promosi.
                  </p>
                </div>
                <div
                  class="MuiFormControl-root inputContainer css-1nrlq1o-MuiFormControl-root"
                  data-testid="inputRHF"
                >
                  <label
                    class="inputLabel"
                    for="name"
                  >
                    Nama Layanan
                    <input
                      class="input undefined"
                      id="name"
                      name="name"
                      type="text"
                      value="Service Name"
                    />
                  </label>
                </div>
              </div>
              <div
                class="inputWrap"
              >
                <div
                  class="inputLabel"
                >
                  <p
                    class="inputLabelTitle"
                  >
                    Harga
                  </p>
                  <p>
                    Harga yang dimasukkan dalam mata uang rupiah (RP.)
                  </p>
                </div>
                <div
                  class="MuiFormControl-root inputContainer css-1nrlq1o-MuiFormControl-root"
                  data-testid="inputRHF"
                >
                  <label
                    class="inputLabel"
                    for="price"
                  >
                    Harga
                    <input
                      class="input undefined"
                      id="price"
                      name="price"
                      type="number"
                      value="50"
                    />
                  </label>
                </div>
              </div>
              <div
                class="inputWrap"
              >
                <div
                  class="inputLabel"
                >
                  <p
                    class="inputLabelTitle"
                  >
                    Layanan Satuan?
                  </p>
                  <p>
                    Tentukan apakah layanan ini satuan atau kiloan
                  </p>
                </div>
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                >
                  <span
                    class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
                  >
                    <span
                      class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                      data-testid="isUnit"
                    >
                      <input
                        checked=""
                        class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                        type="checkbox"
                      />
                      <span
                        class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                      />
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                    />
                  </span>
                </div>
              </div>
              <button
                class="buttonEditProfile"
                data-testid="button-submit"
                type="submit"
              >
                Ubah Layanan
              </button>
            </form>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="mainWrap"
      data-testid="editservice-page"
    >
      <div
        class="header"
      >
        <h3>
          Ubah Layanan
        </h3>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backButton css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="button-back"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowBackIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </span>
          Kembali
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root cardWrap css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root cardContent css-46bh2p-MuiCardContent-root"
        >
          <form>
            <div
              class="inputWrap"
            >
              <div
                class="inputLabel"
              >
                <p
                  class="inputLabelTitle"
                >
                  Nama Layanan
                </p>
                <p>
                  Nama layanan 
Disarankan untuk tidak menggunakan huruf kapital berlebih, memasukkan lebih dari 1 layanan, dan kata-kata promosi.
                </p>
              </div>
              <div
                class="MuiFormControl-root inputContainer css-1nrlq1o-MuiFormControl-root"
                data-testid="inputRHF"
              >
                <label
                  class="inputLabel"
                  for="name"
                >
                  Nama Layanan
                  <input
                    class="input undefined"
                    id="name"
                    name="name"
                    type="text"
                    value="Service Name"
                  />
                </label>
              </div>
            </div>
            <div
              class="inputWrap"
            >
              <div
                class="inputLabel"
              >
                <p
                  class="inputLabelTitle"
                >
                  Harga
                </p>
                <p>
                  Harga yang dimasukkan dalam mata uang rupiah (RP.)
                </p>
              </div>
              <div
                class="MuiFormControl-root inputContainer css-1nrlq1o-MuiFormControl-root"
                data-testid="inputRHF"
              >
                <label
                  class="inputLabel"
                  for="price"
                >
                  Harga
                  <input
                    class="input undefined"
                    id="price"
                    name="price"
                    type="number"
                    value="50"
                  />
                </label>
              </div>
            </div>
            <div
              class="inputWrap"
            >
              <div
                class="inputLabel"
              >
                <p
                  class="inputLabelTitle"
                >
                  Layanan Satuan?
                </p>
                <p>
                  Tentukan apakah layanan ini satuan atau kiloan
                </p>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <span
                  class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
                >
                  <span
                    class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                    data-testid="isUnit"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                      type="checkbox"
                    />
                    <span
                      class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                    />
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                  />
                </span>
              </div>
            </div>
            <button
              class="buttonEditProfile"
              data-testid="button-submit"
              type="submit"
            >
              Ubah Layanan
            </button>
          </form>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
